docs/sdks/server-sdks/python.md

```diff
@@ -51,47 +129,142 @@ analytics.write_key = "<SEGMENT_WRITE_KEY>"
+
+2. Assignments occur within the environments of feature flags. You must **enable the environment** corresponding to the feature flag’s allocation in the [flag control pannel](https://eppo.cloud/feature-flags/) before `getStringAssignment` returns variations. It will return `None` if the environment is not enabled.
+
+![Toggle to enable environment](/static/img/feature-flagging/enable-environment.png)
+
+3. If `get_string_assignment` is invoked **before the SDK has finished initializing**, the SDK may not have access to the most recent experiment configurations. In this case, the SDK will assign a variation based on any previously downloaded experiment configurations stored in local storage, or return `None` if no configurations have been downloaded.
+
+:::info
+
+By default the Eppo client initialization is asynchronous to ensure no critical code paths are blocked. For more information on handling non-blocking initialization, see our [documentation here](/feature-flagging/common-issues#3-not-handling-non-blocking-initialization).
+
+:::
+
+
+## 4. Advanced Configuration with Metadata
+
+We introduced `get_string_assignment`’s two required inputs in the [Getting started](#getting-started) section:
+
+- `subject_key`: The Entity ID that is being experimented on, typically represented by a uuid.
+- `flag_key`: This key is available on the detail page for both flags and experiments.
+
+But that’s not all: the function also takes an optional input for entity properties.
+
+### A. Optional Properties for Targetting 
+
+Most entities on which we run feature flags have properties: sessions have browser types, users have loyalty status, corporate clients have a number of employees, videos have close-caption available or not, sport teams have a league, etc. If you want to decide how a feature flag behaves, or whether an experiment is ran on a certain entity based on those, you need to send that information too. When assigning entities, you can pass that additional information through:
+
+- `subject_attributes`: An optional dictionnary that details entity properties; for example, if the entity is a customer session:  
+  `{country:"Andorra", loyalty:"Gold", browser_type:"Mozilla", device_type:"Macintosh",
+     user_agent:"Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:42.0) Gecko/20100101 Firefox/42.0",}`
+
+Those can be used by the feature flag or the experiement for targeting, through the **Allocations** setting on the configuration page.
+
+### B. Example Payment Configuration
+
+Let’s say you are running a Django service with the User-Agent package. You want to use feature flags to offer payment method that adapt to the browser (only Safari users should be offered to user ApplePay) and the country (so that Dutch users can use iDEAL) and members of your loyalty program might use their points. You can use a feature flag to configure what is possible in which country, for which users, etc. 
```

Comment:
```suggestion
Let’s say you are running a Django service with the User-Agent package. You want to use feature flags to offer payment method that adapt to the browser (only Safari users should be offered to use ApplePay) and the country (so that Dutch users can use iDEAL) and members of your loyalty program might use their points. You can use a feature flag to configure what is possible in which country, for which users, etc. 
```